apkid/rules/apk/obfuscators.yara

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2018  RedNaga. https://rednaga.io
+ * All rights reserved. Contact: rednaga@protonmail.com
+ *
+ *
+ * This file is part of APKiD
+ *
+ *
+ * Commercial License Usage
+ * ------------------------
+ * Licensees holding valid commercial APKiD licenses may use this file
+ * in accordance with the commercial license agreement provided with the
+ * Software or, alternatively, in accordance with the terms contained in
+ * a written agreement between you and RedNaga.
+ *
+ *
+ * GNU General Public License Usage
+ * --------------------------------
+ * Alternatively, this file may be used under the terms of the GNU General
+ * Public License version 3.0 as published by the Free Software Foundation
+ * and appearing in the file LICENSE.GPL included in the packaging of this
+ * file. Please visit http://www.gnu.org/copyleft/gpl.html and review the
+ * information to ensure the GNU General Public License version 3.0
+ * requirements will be met.
+ *
+ **/
+
+include "common.yara"
+
+rule arxan_guardit : obfuscator
+{
+  meta:
+    description = "Arxan GuardIT"
+    url         = "https://www.arxan.com"
+    sample      = ""
+    author      = "Eduardo Novella"
+
+  strings:
+    $cfg = "guardit4j.fin"
```

Comment:
Any way of making this a non-naked string somehow? Maybe it'll be better to fingerprint the dex as well? Rules can reference each other.

Reply:
Nope that I know of. The condition is that there's a file with this name in the APK. We don't know what goes left or right.

Reply:
If it's obfuscating or packing, there should be other stuff to fingerprint other than a single naked string. Let me know if you want to brainstorm ideas.

Reply:
Please go ahead

Reply:
> Any way of making this a non-naked string somehow? Maybe it'll be better to fingerprint the dex as well? Rules can reference each other.

would you like to combine it with Arxan rules? Please @CalebFenton, bring ideas to avoid this naked string. However, the matches were very effective. Check this out at https://koodous.com/rulesets/5078/apks

Reply:
"Please go ahead" does not qualify as a sincere attempt at brainstorming on your end. Please tell me all of the things you've tried that didn't work. We've talked about this many times in the past, and we've talked about different ways of making rules better -- null bytes, decoding opcodes, version strings, etc.